### PR TITLE
Add cookie jar

### PR DESCRIPTION
**Pull request by jokirchner**
I added a cookie jar to http clients used by the docker daemon for login and access to the registry. More details in my comment at #1818

### CODE REVIEW

**Comment by timmjd** [comment]
@manishtomar @dmcgowan @stevvooe could you review?
Another option would be to remove the interface change by making use of a singleton to store the session cookies.

File: registry/client/repository.go [hunk 1]
```diff
@@ -131,7 +139,7 @@ func (r *registry) Repositories(ctx context.Context, entries []string, last stri
 }
 
 // NewRepository creates a new Repository for the given repository name and base URL.
-func NewRepository(name reference.Named, baseURL string, transport http.RoundTripper) (distribution.Repository, error) {
+func NewRepository(name reference.Named, baseURL string, transport http.RoundTripper, jar http.CookieJar) (distribution.Repository, error) {
```
Comment (waynr) [inline registry/client/repository.go:142]: I'm curious why we add a `http.CookieJar` as a parameter on `NewRepository` without adding an instance to it in `proxyingRegistry.Respositories`.

**Comment by milosgajdos** [comment]
@jokirchner can you rebase, please?

**Comment by milosgajdos** [comment]
Ping @johannes-kirchner

**Comment by milosgajdos** [comment]
Closing this PR as in the latest release (https://github.com/distribution/distribution/releases/tag/v3.0.0-alpha.1) we have deprecated the client and made it an `internal` package as there are still some internal dependencies on it. However, it will no longer be exported to external consumers.

Furthermore, we now recommend other clients in the README: https://github.com/distribution/distribution?tab=readme-ov-file#how-does-this-integrate-with-docker-containerd-and-other-oci-client

**Pull request closed without merging**